exe-unit/components/gsb-http-proxy/src/http_to_gsb.rs

```diff
@@ -74,44 +83,52 @@ impl HttpToGsbProxy {
-            BindingMode::Net(binding) => {
-                ya_net::from(binding.from)
-                    .to(binding.to)
-                    .service(&self.bus_addr)
-                    .call(msg)
-                    .await
-            }
-        };
+        let endpoint = self.endpoint();
 
-        let result = response.unwrap_or_else(|e| Err(HttpProxyStatusError::from(e)));
+        log::info!("Proxy http {method} `{path}` call to [{}]", endpoint.addr());
+        let result = endpoint
+            .call(msg)
+            .await
+            .unwrap_or_else(|e| Err(HttpProxyStatusError::from(e)));
 
         match result {
-            Ok(r) => HttpToGsbProxyResponse {
-                body: actix_web::web::Bytes::from(r.body.msg_bytes)
-                    .try_into_bytes()
-                    .map_err(|_| {
-                        Error::GsbFailure("Failed to invoke GsbHttpProxy call".to_string())
-                    }),
-                status_code: r.header.status_code,
-                response_headers: r.header.response_headers,
-            },
-            Err(err) => HttpToGsbProxyResponse {
-                body: actix_web::web::Bytes::from(format!("Error: {}", err))
-                    .try_into_bytes()
-                    .map_err(|_| {
-                        Error::GsbFailure("Failed to invoke GsbHttpProxy call".to_string())
-                    }),
-                status_code: StatusCode::INTERNAL_SERVER_ERROR.as_u16(),
-                response_headers: HashMap::new(),
-            },
+            Ok(r) => {
+                log::debug!(
+                    "Http proxy response for {method} `{path}` call to [{}]: status: {}",
+                    endpoint.addr(),
+                    r.header.status_code
+                );
+                HttpToGsbProxyResponse {
+                    body: actix_web::web::Bytes::from(r.body.msg_bytes)
+                        .try_into_bytes()
+                        .map_err(|_| {
+                            Error::GsbFailure("Failed to invoke GsbHttpProxy call".to_string())
+                        }),
+                    status_code: r.header.status_code,
+                    response_headers: r.header.response_headers,
+                }
+            }
+            Err(err) => {
+                log::warn!(
+                    "Http proxy error calling {method} `{path}` at [{}]: error: {err}",
+                    endpoint.addr()
+                );
+                HttpToGsbProxyResponse {
+                    body: actix_web::web::Bytes::from(format!("Error: {}", err))
+                        .try_into_bytes()
+                        .map_err(|_| {
+                            Error::GsbFailure("Failed to invoke GsbHttpProxy call".to_string())
+                        }),
+                    status_code: StatusCode::INTERNAL_SERVER_ERROR.as_u16(),
+                    response_headers: HashMap::new(),
```

Comment:
Recently Marcin mentioned his worries about not knowing what part of the data path 500 came from.
Maybe we could consider adding some `x-...` header pointing to the yagna proxy's fault?

Reply:
It was considered, but I didn't treat it as part of this task: https://github.com/golemfactory/ya-runtime-ai/issues/109

Reply:
I think you made the correct choice and it was me who was overeager.